models/Site.js

```diff
@@ -30,7 +21,14 @@ const siteSchema = new mongoose.Schema({
   isPrivate: Boolean,
   members: [mongoose.Schema.Types.ObjectId], // It might not be needed in near future
   navigations: [navigationSchema],
-  pages: [pageSchema],
+  pages: [
```

Comment:
Can't we just do `[{ type: mongoose.Schema.Types.ObjectId, ref: 'Page' }]`. Any reason why you want to do it like that?